src/mbgl/util/tile_cover.cpp

```diff
@@ -85,7 +69,7 @@ std::vector<UnwrappedTileID> tileCover(const Point<double>& tl,
                                        const Point<double>& bl,
                                        const Point<double>& c,
                                        int32_t z) {
-    const int32_t tiles = 1 << z;
+    const int32_t tiles = (1 << z) + 1;
 
     struct ID {
```

Comment:
We can add constructor and then on line 85
`t.emplace_back(x, y, dx * dx + dy * dy);`

I quickly tested, old version takes 21ns per insertion, with proper emplace 14ns.

Reply:
Thanks. Splitting the tile cover from patch so I'll integrate it there.